bubblegum/js/tests/main.test.ts

```diff
@@ -204,6 +206,72 @@ describe('Bubblegum tests', () => {
       console.log('Verified NFT existence:', txId);
     });
 
+    it('Can transfer and burn a compressed NFT', async () => {
+      const accountInfo = await connection.getAccountInfo(merkleTree, { commitment: 'confirmed' });
+      const account = ConcurrentMerkleTreeAccount.fromBuffer(accountInfo!.data!);
+      const [treeAuthority, _bump] = await PublicKey.findProgramAddress(
+        [merkleTree.toBuffer()],
+        BUBBLEGUM_PROGRAM_ID,
+      );
+      const newLeafOwnerKeypair = new Keypair();
+      const newLeafOwner = newLeafOwnerKeypair.publicKey;
+
+      const transferIx = createTransferInstruction(
+        {
+          treeAuthority,
+          leafOwner: payer,
+          leafDelegate: payer,
+          newLeafOwner,
+          merkleTree,
+          logWrapper: SPL_NOOP_PROGRAM_ID,
+          compressionProgram: SPL_ACCOUNT_COMPRESSION_PROGRAM_ID,
+        },
+        {
+          root: Array.from(account.getCurrentRoot()),
+          dataHash: Array.from(computeDataHash(originalCompressedNFT)),
+          creatorHash: Array.from(computeCreatorHash(originalCompressedNFT.creators)),
+          nonce: 0,
+          index: 0
+        },
+      );
+
+      const transferTx = new Transaction().add(transferIx);
+      transferTx.feePayer = payer;
+      const transferTxId = await sendAndConfirmTransaction(connection, transferTx, [payerKeypair], {
+        commitment: 'confirmed',
+        skipPreflight: true,
+      });
+
+      console.log('Verified NFT transfer:', transferTxId);
+
+      const burnIx = createBurnInstruction(
+        {
+          treeAuthority,
+          leafOwner: newLeafOwner,
+          leafDelegate: newLeafOwner,
+          merkleTree,
+          logWrapper: SPL_NOOP_PROGRAM_ID,
+          compressionProgram: SPL_ACCOUNT_COMPRESSION_PROGRAM_ID,
+        },
+        {
+          root: Array.from(account.getCurrentRoot()),
+          dataHash: Array.from(computeDataHash(originalCompressedNFT)),
+          creatorHash: Array.from(computeCreatorHash(originalCompressedNFT.creators)),
```

Comment:
Please review this, it works but is it the best way to specify these params?

Reply:
This looks good to me. What are you concerned about with these array parameters?